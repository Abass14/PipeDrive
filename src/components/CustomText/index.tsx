import React from "react";
import { StyleProp, Text, TextProps, TextStyle } from "react-native";

type CustomTextProps = {
    children: string | React.ReactNode,
    style?: StyleProp<TextStyle>,
    textProps?: TextProps,
    bold?: boolean
}
const CustomText: React.FC<CustomTextProps> = ({
    children,
    style,
    textProps,
    bold
}) => {
    return (
        <Text style={[{
            fontSize: 12,
            fontWeight: bold ? '700' : 'normal'
        }, style]} {...textProps}>
            {children}
        </Text>
    )
}

export default CustomText