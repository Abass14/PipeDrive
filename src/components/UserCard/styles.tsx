import React from "react";
import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
    container: {
        width: '100%',
        borderRadius: 15,
        padding: 15,
        elevation: 2,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        backgroundColor: 'white',
    },
    innerConterOne: {
        flex: 1,
        flexDirection: 'row'
    },
    textContainer: {
        marginLeft: 10, 
        justifyContent: 'center',
        alignItems: 'flex-start'
    },
    innerContainerTwo: {
        justifyContent: 'center',
        alignItems: 'flex-end'
    },
    avatarContainer: {
        width: 60,
        height: 60,
        overflow: 'hidden',
        resizeMode: 'cover',
        borderRadius: 30,
    },
    avatar: {
        width: 60,
        height: 60,
        resizeMode: 'cover',
    },
    names: {
        justifyContent: 'center',
        alignItems: 'flex-end',
        maxWidth: 200
    },
})