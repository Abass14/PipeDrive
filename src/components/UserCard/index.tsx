import React from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";
import { capitalize } from "../../utils/textUtils";
import CustomText from "../CustomText";
import IconView from "../IconView";
import { styles } from "./styles";

type UserCardProps = {
    imageUri: string,
    userName: string,
    email: string,
    onPress?: () => void
}
const UserCard: React.FC<UserCardProps> = ({
    imageUri,
    userName,
    email,
    onPress
}): JSX.Element => {

    return (
        <TouchableOpacity onPress={onPress} style={styles.container}>
            <View style={styles.innerConterOne}>
                <View style={styles.avatarContainer}>
                    <Image source={{ uri: imageUri }} style={styles.avatar} resizeMode={"cover"} />
                </View>
                <View style={styles.textContainer}>
                    <CustomText textProps={{numberOfLines: 1}} bold={true} style={[styles.names, { fontSize: 16 }]}>
                        {capitalize(userName)}
                    </CustomText>
                    <CustomText style={styles.names} textProps={{numberOfLines: 1}}>
                        {email}
                    </CustomText>
                </View>
            </View>
            <View style={styles.innerContainerTwo}>
                <IconView
                    type="material"
                    name="chevron-right"
                    size={30}
                />
            </View>
        </TouchableOpacity>
    )
}

export default UserCard;