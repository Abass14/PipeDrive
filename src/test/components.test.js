import HyperLink from "../components/Hyperlink";

import renderer from "react-test-renderer";
import CustomText from "../components/CustomText";
import UserCard from "../components/UserCard";

it('renders hyperlink correctly', () => {
  const tree = renderer
    .create(<HyperLink text={"text"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders custom text correctly', () => {
  const tree = renderer
    .create(<CustomText>Hello</CustomText>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});