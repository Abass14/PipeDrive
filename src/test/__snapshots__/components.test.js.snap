// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom text correctly 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12,
        "fontWeight": "normal",
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;

exports[`renders hyperlink correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 12,
          "fontWeight": "700",
        },
        [
          {
            "textDecorationLine": "underline",
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;
